plot_num_threads_per_disk.py

```diff
@@ -0,0 +1,183 @@
+      num_threads = __get_num_threads_from_log_dir(trial_log_dir)
+      jcts = __get_jcts_from_logs(trial_log_dir_filepath, args.warmup_count)
+      num_threads_to_jcts[num_threads] = jcts
+
+  assert len(num_threads_to_jcts) > 0, "No valid logs found in %s" % log_dir
+
+  __create_num_threads_vs_jct_graph(num_threads_to_jcts, args.output_dir, phase="write")
+  __create_num_threads_vs_jct_graph(num_threads_to_jcts, args.output_dir, phase="read")
+
+
+def __parse_args():
+  parser = argparse.ArgumentParser(description="")
+  parser.add_argument(
+    "-l",
+    "--log-dir",
+    help=("A directory that contains several directories of logs. Each log " +
+      "subdirectory should contain the results from one run of " +
+      "org.apache.spark.examples.monotasks.disk.DiskThroughputExperiment " +
+      "and have been generated by the utils.copy_and_zip_all_logs() function."),
+    required=True)
+  parser.add_argument(
+    "-o",
+    "--output-dir",
+    help="The directory in which to store the graph PDFs. Defaults to the value of --log-dir.",
+    required=False)
+  parser.add_argument(
+    "-w",
+    "--warmup-count",
+    default=0,
+    help="The number of iterations that are warmup and should be discarded.",
+    required=True,
+    type=int)
+
+  args = parser.parse_args()
+  log_dir = args.log_dir
+  assert path.isdir(log_dir), "The supplied log directory does not exist or is a file: %s" % log_dir
+
+  if args.output_dir is None:
+    args.output_dir = args.log_dir
+  return args
+
+
+def __plot_continuous_monitors(log_dir):
+  """ Plots all of the continuous monitors in the provided directory. """
+  for log_filename in os.listdir(log_dir):
+    if log_filename.endswith("executor_monitor"):
+      plot_continuous_monitor.plot_continuous_monitor(
+        path.join(log_dir, log_filename), use_gnuplot=True)
+
+
+def __get_jcts_from_logs(log_dir, warmup_count):
+  """
+  Returns a tuple of (list of write job JCTs, list of read job JCTs) parsed from the event log
+  contained in the provided directory.
+  """
+  event_log_filepath = path.join(log_dir, "event_log")
+  return (__get_jcts_from_logs_for_phase(event_log_filepath, warmup_count, phase="write"),
+    __get_jcts_from_logs_for_phase(event_log_filepath, warmup_count, phase="read"))
+
+
+def __get_jcts_from_logs_for_phase(event_log_filepath, warmup_count, phase):
+  """
+  Returns a list of JCTs (in seconds) for the specified phase parsed from the specified event log.
+  warmup_count indicates the number of jobs that should be discarded. Assumes that the event log
+  corresponds to one run of org.apache.spark.examples.monotasks.disk.DiskThroughputExperiment.
+  """
+  assert phase in ["write", "read"]
+  filterer = __write_job_filterer if phase == "write" else __read_job_filterer
+  filterer_partial = functools.partial(filterer, warmup_count)
+  return [job.runtime() / 1000.0 for job in
+    parse_event_logs.Analyzer(event_log_filepath, filterer_partial).jobs.itervalues()]
```

Comment:
would it make sense to just run Analyzer once, and then pass in the [(job id, job)] list to this function? (since making the analayzer by parsing everything can be kind of slow)